fedimint-core/src/lib.rs

```diff
@@ -250,38 +250,58 @@ pub enum ParseAmountError {
     WrongBitcoinAmount(#[from] bitcoin::util::amount::ParseAmountError),
 }
 
-impl<T> NumPeers for BTreeMap<PeerId, T> {
+impl<T> NumPeersExt for BTreeMap<PeerId, T> {
     fn total(&self) -> usize {
         self.len()
     }
 }
 
-impl NumPeers for &[PeerId] {
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct NumPeers(usize);
+
+impl From<usize> for NumPeers {
+    fn from(value: usize) -> Self {
+        Self(value)
+    }
+}
+impl NumPeers {
+    pub fn as_usize(self) -> usize {
```

Comment:
```suggestion
    pub fn as_usize(&self) -> usize {
```

Reply:
`NumPeers` is `Copy`, so no need to `&`

Reply:
yeah, but it's ugly an `as_*` method taking `self`

Reply:
Good point.